### ISSUE TITLE
[Paladin] Auto Attack when spam casting Flash of Light = Infinite Mana.

### ISSUE THREAD

**Issue by mrnuggets** [dalaranwow/dalaran-wow#5828]
[//]: # (INFORMATION ABOUT TAGS)
[//]: # (The black and yellow tag is unique. There can only be one label assigned to them - one black and one yellow)
[//]: # (Priority - ... tags are assigned by team members)
[//]: # (Your issue will be reviewed by testers and the developers will fix it in the future.)
[//]: # (Only once the issue got the label [Fixed - On Live Server] you can find the fix on your Gameserver.)
[//]: # (You can increase the relevance of your issue by adding a thumb-up emote http://i.imgur.com/RFouVVi.png)

[//]: # (***************************************************************)
[//]: # (** !!!!!!!!!!!!!DON'T DELETE THIS TEMPLATE ELSE YOUR ISSUE WILL BE CLOSED!!!!!!!!!!!! **)
[//]: # (***************************************************************)

**Current behaviour**: Tell us what happens in detail (Where, When, What, Why, Who)

By spamming Flash of Light, you can have infinite mana, cause you will auto-attack the target on every new cast Flash of Light, and regain the mana.

**Expected behaviour**: Tell us what should happen instead

It should not auto-attack on each cast.

**Steps to reproduce the problem**:

1. Attack the boss.
2. Spam Flash of Light.
3. Watch as you will attack the target every time a new Flash of Light is cast.

**Include proofs for this behaviour**

https://youtu.be/cAYCqeBa5fc

You can research this on several pages like http://wowhead.com or http://wowwiki.wikia.com or 

http://www.youtube.com/

Issues without proofs will be closed and tag On Hold will be added until the proof is provided.

**Include the ID for the game objects, npcs (creatures,pets,minions), spells, items, quests, instances, zones, achievements, skills**

Spell ID: 48784

You can research this on several pages like https://wowgaming.altervista.org/aowow/ or https://wowhead.com/

**Include Screenshots from the issue if necessary**

**Comment by Chenocide** [comment]
Technically this has nothing to do with Flash of Light, as any spell has the same issue, including other classes. You can benefit from Judgement of Wisdom simply standing in melee range as a druid, shaman or priest and you will regenerate mana in between your spell casts.

**Comment by mrnuggets** [comment]
> Technically this has nothing to do with Flash of Light, as any spell has the same issue, including other classes. You can benefit from Judgement of Wisdom simply standing in melee range as a druid, shaman or priest and you will regenerate mana in between your spell casts.

To refute your point given here, as you can see in the video, Judgement of Light is used.

Here's a video of me using Judgement of Light and Judgement of Wisdom, standing close to the target, and as you can see, my mana is dropping significantly without auto-attacking, while if i was auto-attacking, i wouldn't lose mana.

**Comment by Kennehsauce** [comment]
Are you saying the auto attack shouldn't go off because you think you're spamming faster than the swing timer?

**Comment by mrnuggets** [comment]
> Are you saying the auto attack shouldn't go off because you think you're spamming faster than the swing timer?

It's as evident as day. When you're SPAMMING a spell, you should NOT be able to auto attack. If you watch the video, you will also see me attacking the dummy more than once while casting.

**Comment by Kennehsauce** [comment]
> 
> 
> > Are you saying the auto attack shouldn't go off because you think you're spamming faster than the swing timer?
> 
> It's as evident as day. When you're SPAMMING a spell, you should NOT be able to auto attack. If you watch the video, you will also see me attacking the dummy more than once while casting.
> 
> But to me this doesn't matter at this point. Posted this months ago, and it's never getting fixed.

Not quite, if your cast speed is faster than the GCD and your swing speed is just under it the GCD as well, it stands to reason that it would melee regardless because there's a split second for it to happen.

GCD =1.5 sec
Flash = roughly .9 at your GS
Swing speed =1~
That's a whole .5~ for it to go off regardless if you're spamming it, no?

If you ditch all you're gear and try it you still manage to do it but its not on every cast.

Am I missing something here fam? Doesn't seem broken at all.

**Comment by Chenocide** [comment]
> > > Are you saying the auto attack shouldn't go off because you think you're spamming faster than the swing timer?
> > 
> > 
> > It's as evident as day. When you're SPAMMING a spell, you should NOT be able to auto attack. If you watch the video, you will also see me attacking the dummy more than once while casting.
> > But to me this doesn't matter at this point. Posted this months ago, and it's never getting fixed.
> 
> So I think you mean, Casting, in general, should reset your swing timer. Therefore, you shouldn't be able to melee at all. The bug was already addressed [here](https://github.com/dalaranwow/dalaran-wow/issues/4679) and you're right, they won't fix it lel.

If you spam an ability, then that ability should be queued for immediate use and you should not be able to auto attack in between an ability and a queued ability.  Even if the linked report was fixed, you would still be able to get in auto attacks if the cast time of the spell was more than 1.5 seconds, hence why this is a separate problem but in the same underlying mechanic.
